app/interactors/backdate/update_interactor.rb

```diff
@@ -0,0 +1,70 @@
+# frozen_string_literal: true
+
+class Backdate::UpdateInteractor
+  include Interactor
+  delegate :params,
+           :user,
+           :edition,
+           :date,
+           to: :context
+
+  def call
+    Edition.transaction do
+      find_and_lock_edition
+      check_if_first_edition
+
+      parse_date
+      check_for_issues
+
+      update_edition
+      create_timeline_entry
+    end
+  end
+
+private
+
+  def find_and_lock_edition
+    context.edition = Edition.lock.find_current(document: params[:document])
+  end
+
+  def check_if_first_edition
```

Comment:
Minor - I think we now tend to do this in the 'find' method, as it's really just an assertion about the edition that we've found e.g. https://github.com/alphagov/content-publisher/blob/master/app/interactors/schedule/create_interactor.rb#L25

Reply:
I think @kevindew is also keen on an 'editable?' check, although we're not really consistent on this.

https://trello.com/c/eAGtt4ry/322-we-dont-enforce-our-view-restrictions-in-our-controllers

Reply:
Makes sense, we do check if the edition is editable before rendering the context settings section so we should make that consistent here.

Reply:
Yup 👍 looking this up in find the most consistent place we have.